src/lib/core/proc_name_cache.cc

```diff
@@ -0,0 +1,107 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#include "fiber.h"
+#include "proc_name_cache.h"
+
+#include "small/region.h"
+
+enum {
+	PROC_NAME_MAX = 64,
+};
+
+/* Procedure name hash table entry. */
+struct proc_name_cache_entry {
+	/* Demangled procedure name. */
+	char name[PROC_NAME_MAX];
+	/* Procedure offset. */
+	unw_word_t offset;
+};
+
+#define mh_name _proc_name_cache
+#define mh_key_t unw_word_t
+
+/* Procedure name hash table node. */
+struct mh_proc_name_cache_node_t {
+	/* Procedure RIP value. */
+	mh_key_t key;
+	/* Procedure name and offset. */
+	struct proc_name_cache_entry *entry;
+};
+
+#define mh_node_t struct mh_proc_name_cache_node_t
+#define mh_arg_t void *
+#define mh_hash(a, arg) ((a)->key)
+#define mh_hash_key(key, arg) (key)
+#define mh_cmp(a, b, arg) ((a)->key != (b)->key)
+#define mh_cmp_key(key, b, arg) ((key) != (b)->key)
+#define MH_SOURCE
+#include "salad/mhash.h"
```

Comment:
Why not reuse `mh_i64ptr_t`? It looks like in the old version of the cache we used `mh_i64ptr_t` and didn't define a new hash.

Reply:
Fixed: reused `mh_i64ptr_t`